README.md

```diff
@@ -98,7 +98,12 @@ See the [examples](./examples) directory for different integration patterns.
 
 ## Overview of crates
 
-The following crates are maintained in this repo:
+OpenTelemetry Rust is composed of multiple interdependent crates as given below, 
+each of which may have different version numbers. To maintain compatibility 
+and ensure proper functionality, it is recommended to update all related 
+crates simultaneously when new versions are released.  Although it would be 
+ideal to avoid breaking changes with minor version updates, this is currently 
```

Comment:
This makes it sound like are we are required to avoid such breaking changes, but we would not be able to do it. It's completely valid under Semver compatibility rules to make breaking changes with minor version updates for `0.x.x` versions.

If the intention is to warn users who might have misconception about what kind of breaking changes are allowed, maybe we could just reiterate the relevant Semver compatibility rule. Something similar to the text in this [doc](https://doc.rust-lang.org/cargo/reference/resolver.html#semver-compatibility):

> Versions are considered compatible if their left-most non-zero major/minor/patch component is the same. For example, 0.1.0 and 0.1.2 are compatible, but 0.1.0 and 0.2.0 are not.

Reply:
> It's completely valid under Semver compatibility rules to make breaking changes with minor version updates for 0.x.x versions.

This is what I read from [semver](https://semver.org/):
```
MAJOR version when you make incompatible API changes
MINOR version when you add functionality in a backward compatible manner
PATCH version when you make backward compatible bug fixes
```

Reply:
Oh you mean with `0` as major version.

Reply:
The issue I observe is that in the Rust ecosystem, most of crates are in 0.x.x versions but are considered stable, with minimal breaking changes occurring with minor version updates. I just wanted to emphasize that for OpenTelemetry, this would be comparatively more frequent.

Reply:
Got it. Thanks for providing the context! Maybe we can put this text under its own section in the doc to highlight this point even more. Something like this: https://github.com/rust-random/rand?tab=readme-ov-file#versions

Reply:
I am unsure what is the issue we are trying to address with this wording? Can you elaborate.
Is it to convey that we are not stable?
Is it to convey that breaking changes are possible within minor version bump, until a crate hits 1.0?
Or more?

Reply:
> I am unsure what is the issue we are trying to address with this wording? Can you elaborate.
> Is it to convey that we are not stable?
> Is it to convey that breaking changes are possible within minor version bump, until a crate hits 1.0?
> Or more?

Maybe I haven't put the wordings properly. The main emphasis was this

> To maintain compatibility  and ensure proper functionality, it is recommended to update all related 
crates simultaneously when new versions are released. 

 i.e., to ask users to update to the latest versions of all OpenTelemetry crates together when new releases are made. Additionally, if there are any indirect dependencies on OpenTelemetry crates (e.g. such as those coming through `tracing-opentelemetry`), they should ensure that the versions are same (preferably latest). If not, reach out to the repo providing this dependency asking them to update.

But looks like the PR is causing more confusion, and less value.  Would be good to close this :)